openlibrary/plugins/openlibrary/api.py

```diff
@@ -151,9 +151,9 @@ def POST(self, work_id):
                 'error': 'Invalid bookshelf'
             }), content_type="application/json")
 
-        if bookshelf_id == current_status:
+        if bookshelf_id == current_status or bookshelf_id == -1:
```

Comment:
Don't need to support that anymore

```suggestion
        if bookshelf_id == -1:
```

Reply:
Unfortunately it looks like the `widget.html` file is still using this logic (searching for `/bookshelves.json` revealed). I think it also relies on `?action=remove` doing something (which it does not), so there's also that :/

Reply:
Won't fix